server/src/test/java/org/elasticsearch/monitor/jvm/HotThreadsTests.java

```diff
@@ -307,13 +307,31 @@ public void testEnsureInnerDetectSkipsCurrentThread() throws Exception {
 
         HotThreads hotThreads = new HotThreads()
             .busiestThreads(4)
-            .type("cpu")
+            .type(HotThreads.ReportType.CPU)
             .interval(TimeValue.timeValueNanos(10))
             .threadElementsSnapshotCount(11)
             .ignoreIdleThreads(false);
 
-        var innerResult = hotThreads.innerDetect(mockedMXBean, mockCurrentThreadId);
+        String innerResult = hotThreads.innerDetect(mockedMXBean, mockCurrentThreadId);
```

Comment:
Are these type changes to make the backport easier?

Reply:
Yeah, I started using inferred types, but makes extending the tests hard for 7.x and I plan to add few more unit tests.